kv/memberlist/tcp_transport.go

```diff
@@ -91,7 +99,11 @@ type TCPTransport struct {
 	tcpListeners []net.Listener
 	tlsConfig    *tls.Config
 
-	shutdown atomic.Int32
+	writeCh chan writeRequest
+	writeWG sync.WaitGroup
```

Comment:
I'd suggest to move this under `shutdownMu`, since sends to channel are done with read-lock and channel closing is done with write-lock (also worth mentioning in the comment).

Reply:
Sorry, I meant the channel, not `WaitGroup`.